model/rulefmt/rulefmt.go

```diff
@@ -96,7 +96,14 @@ type ruleGroups struct {
 }
 
 // Validate validates all rules in the rule groups.
-func (g *RuleGroups) Validate(node ruleGroups) (errs []error) {
+func (g *RuleGroups) Validate(node ruleGroups, nameValidationScheme model.ValidationScheme) (errs []error) {
+	switch nameValidationScheme {
+	case model.UTF8Validation, model.LegacyValidation:
+	default:
+		errs = append(errs, fmt.Errorf("unhandled nameValidationScheme: %s", nameValidationScheme))
+		return
+	}
```

Comment:
this pattern shows up a lot, I wonder if we can avoid repeating it so much? Not a blocker for this PR